### PR TITLE
adding Alaska Medical Assistance Excluded providers crawler

### PR DESCRIPTION
**Pull request by bgmello**

### CODE REVIEW

**Comment by bgmello** [comment]
There where some GPT couldn't get the name, like the one below:

Screenshot 2024-09-16 at 15 39 07

I guess because the name is slightly cut, so I just ignored those.

File: datasets/us/ak/med_exclusions/us_ak_med_exclusions.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  The Alaska Department of Health publishes a list of Medical provider
+  who have been excluded from Alaska Medical Assistance Program.
+  > Exclusions are the result of criminal, administrative or civil
+  > adverse actions and are publically noticed at the time of exclusion.
+publisher:
+  name: Alaska Department of Health
+  url: https://health.alaska.gov/Pages/default.aspx
+  official: true
+  country: "us"
+url: "https://health.alaska.gov/Commissioner/Documents/PDF/AlaskaExcludedProviderList.pdf"
```
Comment (jbothma) [inline datasets/us/ak/med_exclusions/us_ak_med_exclusions.yaml:20]:
```suggestion
url: https://health.alaska.gov/Commissioner/Pages/ProgramIntegrity/default.aspx
```

File: datasets/us/ak/med_exclusions/us_ak_med_exclusions.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+load_db_uri: ${OPENSANCTIONS_DATABASE_URI}
+summary: >
+  Individuals and business excluded from Alaska Medical Assistance Program
+description: |
+  The Alaska Department of Health publishes a list of Medical provider
+  who have been excluded from Alaska Medical Assistance Program.
+  > Exclusions are the result of criminal, administrative or civil
+  > adverse actions and are publically noticed at the time of exclusion.
+publisher:
+  name: Alaska Department of Health
```
Comment (jbothma) [inline datasets/us/ak/med_exclusions/us_ak_med_exclusions.yaml:16]: description
Perhaps AK-DOH as acronym

File: datasets/us/ak/med_exclusions/crawler.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    if not row.get("last_name"):
+        return
+
+    if not row.get("first_name"):
+        entity = context.make("Company")
+        entity.id = context.make_id(row.get("last_name"))
+        entity.add("name", row.pop("last_name"))
+    else:
+        entity = context.make("Person")
+        entity.id = context.make_id(row.get("last_name"), row.get("first_name"))
```
Comment (jbothma) [inline datasets/us/ak/med_exclusions/crawler.py:28]: let's add date

File: datasets/us/ak/med_exclusions/crawler.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        h.apply_name(
+            entity, first_name=row.pop("first_name"), last_name=row.pop("last_name")
+        )
+
+    entity.add("sector", row.pop("provider_type"))
```
Comment (jbothma) [inline datasets/us/ak/med_exclusions/crawler.py:33]: let's also add country

File: datasets/us/ak/med_exclusions/crawler.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+from rigour.mime.types import PDF
+
+from zavod import Context, helpers as h
+
+from zavod.shed.gpt import run_image_prompt
+
+prompt = """
+Extract structured data from the following page of a PDF document. Return 
+a JSON list (`providers`) in which each object represents an medical provider.
+Each object should have the following fields: `exclusion_date`, `last_name`,
```
Comment (jbothma) [inline datasets/us/ak/med_exclusions/crawler.py:11]: This seems to fix the example you showed. Does it fix any others that you found to have issues? Does it introduce any other issues?

```suggestion
a JSON list (`providers`) in which each object represents an medical provider.
Distinct record rows alternate between grey and white backgrounds. The name
of a provider might be split onto the next row within a record.
Each object should have the following fields: `exclusion_date`, `last_name`,
```

I am absolutely blown away that this works. So interesting.

**Review by jbothma — COMMENTED** [review]
Awesome. Just about done. Small tweaks and we ship.

File: datasets/us/ak/med_exclusions/crawler.py (hunk 5 above)
Comment (bgmello) [reply datasets/us/ak/med_exclusions/crawler.py]: I was too! It almost look like magic.

Apparently this doesn't introduce any new issues, I ran and it didn't enter the `not row.get("last_name")` if statement.

**Pull request merged**